linera-core/src/worker.rs

```diff
@@ -753,30 +753,24 @@ where
     async fn get_chain_worker_endpoint(
         &self,
         chain_id: ChainId,
-    ) -> Result<ChainActorEndpoint<StorageClient>, WorkerError> {
-        let mut chain_workers = self.chain_workers.lock().await;
-
-        if !chain_workers.contains(&chain_id) {
-            chain_workers.push(
-                chain_id,
+    ) -> ChainActorEndpoint<StorageClient> {
+        self.chain_workers
```

Comment:
We could spawn the actor here. Something like (didn't test this):

```rust
let mut new_receiver = None;

let endpoint = self.chain_workers
    .lock()
    .expect("`WorkerState` should not panic while accessing the `chain_workers` map")
    .get_or_insert(chain_id, || {
        let (sender, receiver) = mpsc::unbounded_channel();
        new_receiver = Some(receiver);
        sender
    })
    .clone();

if let Some(receiver) = new_receiver {
    let actor = ChainWorkerActor::new(
            self.chain_worker_config.clone(),
            self.storage.clone(),
            self.recent_hashed_certificate_values.clone(),
            self.recent_hashed_blobs.clone(),
            chain_id,
            receiver,
        ).await?;
    self.chain_worker_tasks
        .lock()
        .expect("`WorkerState` should not panic while accessing `chain_worker_tasks`")
        .spawn_task(actor.run(tracing::Span::current()));
}

endpoint
```

Reply:
I already started doing this on my local branch :)  Great minds think alike!